### DISCUSSION TITLE
How is createTime and updateTime set?

### DISCUSSION THREAD

**Discussion by bhaskarmurthy** [tylim88/FirelordJS#105] [Q&A] [answered]
I'm trying to understand how the `createTime` and `updateTime` values are set.

When I use the `addDoc` method, it seems to automatically generate the timestamp - which is accessible in the `createTime` and `updateTime` fields from the snapshot.

However - when I go to Firestore, there are no such fields in the document.
Where is this value stored?

Example:
```
type Organization = {
  name: string;
};
type OrganizationMeta = MetaTypeCreator<Organization, 'organizations'>;

const db = getFirestore();
const firelord = getFirelord<OrganizationMeta>(db, 'organizations');
const organization: Organization = { name: "Hello, world" };
const ref = await addDoc(firelord.collection(), organization);
const snap = await getDoc(firelord.doc(ref.id));

console.log('createTime', snap.createTime); // this is somehow set
console.log('updateTime', snap.updateTime); // this is somehow set
console.log('data', snap.data()); // this does not contain timestamp properties

// also checking Firestore console, there are no timestamp properties in the document
```

Thanks!

**Comment by tylim88** [comment]
Firelord doesnt set them, in fact this is the first time i see something like these

They seem to be meta data, that is why you cant find them in data()

I think they are hidden API of official SDK because they cant be found in official documentation https://firebase.google.com/docs/reference/node/firebase.firestore.DocumentSnapshot

**Comment by tylim88** [comment, marked as answer]
update: I found them

https://cloud.google.com/nodejs/docs/reference/firestore/latest/firestore/documentsnapshot#_google_cloud_firestore_DocumentSnapshot_createTime_member

they can only be accessed from admin sdk, they are meta data, not part of your data

**Comment by bhaskarmurthy** [comment]
This is amazing - never knew this existed. Thanks for looking into this!
